ui/ducks/bridge/actions.ts

```diff
@@ -62,3 +166,359 @@ export const setToChain = (chainId: Hex) => {
+        .toString();
+
+      const shouldResetApproval = allowance.lt(sentAmount) && allowance.gt(0);
+
+      if (shouldResetApproval) {
+        const resetData = getEthUsdtResetData();
+        const txParams = {
+          ...approval,
+          data: resetData,
+        };
+
+        await handleTx({
+          txType: 'bridgeApproval',
+          txParams,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          meta: {
+            type: 'bridgeApproval', // TransactionType.bridgeApproval,
+          },
+        });
+      }
+    };
+
+    const handleApprovalTx = async ({
+      approval,
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+    }: {
+      approval: TxData;
+      maxFeePerGas: string | undefined;
+      maxPriorityFeePerGas: string | undefined;
+    }) => {
+      const hexChainId = new Numeric(
+        approval.chainId,
+        10,
+      ).toPrefixedHexString() as `0x${string}`;
+
+      // On Ethereum, we need to reset the allowance to 0 for USDT first if we need to set a new allowance
+      // https://www.google.com/url?q=https://docs.unizen.io/trade-api/before-you-get-started/token-allowance-management-for-non-updatable-allowance-tokens&sa=D&source=docs&ust=1727386175513609&usg=AOvVaw3Opm6BSJeu7qO0Ve5iLTOh
+      if (isEthUsdt(hexChainId, quoteResponse.quote.srcAsset.address)) {
+        await handleEthUsdtAllowanceReset({
+          approval,
+          hexChainId,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+        });
+      }
+
+      const txMeta = await handleTx({
+        txType: 'bridgeApproval',
+        txParams: approval,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+        meta: {
+          type: 'bridgeApproval', // TransactionType.bridgeApproval, // TODO
+          sourceTokenSymbol: quoteResponse.quote.srcAsset.symbol,
+        },
+      });
+
+      return txMeta.id;
+    };
+
+    const handleBridgeTx = async ({
+      approvalTxId,
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+    }: {
+      approvalTxId: string | undefined;
+      maxFeePerGas: string | undefined;
+      maxPriorityFeePerGas: string | undefined;
+    }) => {
+      const txMeta = await handleTx({
+        txType: 'bridge',
+        txParams: quoteResponse.trade,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+        meta: {
+          // estimatedBaseFee: decEstimatedBaseFee,
+          // swapMetaData,
+          type: 'bridge', // TransactionType.bridge, // TODO add this type
+          sourceTokenSymbol: quoteResponse.quote.srcAsset.symbol,
+          destinationTokenSymbol: quoteResponse.quote.destAsset.symbol,
+          destinationTokenDecimals: quoteResponse.quote.destAsset.decimals,
+          destinationTokenAddress: quoteResponse.quote.destAsset.address,
+          approvalTxId,
+          // this is the decimal (non atomic) amount (not USD value) of source token to swap
+          swapTokenValue: new Numeric(quoteResponse.quote.srcTokenAmount, 10)
```

Comment:
I think we should use the `sentAmount` to include the fees?

Reply:
Done in [9017cd9](https://github.com/MetaMask/metamask-extension/pull/27262/commits/9017cd981cbc3732c858045168672db6e96c9dfe)